File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/player/DisplayNameService.java
package top.likoslupus.cellulosesz.api.player;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;
import top.likoslupus.cellulosesz.api.text.RichText;

import java.util.UUID;

public interface DisplayNameService {

    RichText displayName(CellPlayer player);

    RichText displayName(UUID uuid, String fallbackName);

    String plainDisplayName(CellPlayer player);

    boolean validNickname(CellPlayer player, String nickname);

    String sanitizeNickname(CellPlayer player, String nickname);

    void refresh(CellPlayer player);

    void refreshAll();

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/player/DisplayNameService.kt
package top.likoslupus.cellulosesz.api.player

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import top.likoslupus.cellulosesz.api.text.RichText
import java.util.*

public interface DisplayNameService {

    public fun displayName(player: CellPlayer): RichText

    public fun displayName(uuid: UUID, fallbackName: String): RichText

    public fun plainDisplayName(player: CellPlayer): String

    public fun validNickname(player: CellPlayer, nickname: String): Boolean

    public fun sanitizeNickname(player: CellPlayer, nickname: String): String

    public fun refresh(player: CellPlayer)

    public fun refreshAll()

}
